tsvtools/ocrd_processors.py

```diff
@@ -0,0 +1,108 @@
+        noproxy = self.parameter['noproxy']
+
+        ppn_found = self.workspace.mets._tree.find('//mods:recordIdentifier[@source="gbv-ppn"]', NS)
+        print(ppn_found)
+        if ppn_found is not None:
+            ppn = ppn_found.text
+        else:
+            ppn = ''
+        for n, input_file in enumerate(self.input_files):
+            page_id = input_file.pageId or input_file.ID
+            log.info('Processing: %d / %s of %d', n, page_id, len(list(self.input_files)))
+            file_id = make_file_id(input_file, self.output_file_grp)
+            pcgts = page_from_file(self.workspace.download_file(input_file))
+            page = pcgts.get_Page()
+
+            iiif_url = iiif_url_template\
+                    .replace('{{ unique_identifier }}', self.workspace.mets.unique_identifier)\
+                    .replace('{{ PPN }}', ppn)\
+                    .replace('{{ page_id }}', page_id)\
+                    .replace('{{ page_no }}', re_sub('[^0-9]', '', page_id))
+            Path(self.output_file_grp).mkdir(exist_ok=True)
+            tsv_filepath = Path(self.output_file_grp, file_id + '.tsv')
+            page2tsv(input_file.local_filename, tsv_filepath, 'OCR', iiif_url, None, None, noproxy, 1.0, None, None, None, 1)
+
+            self.workspace.add_file(
+                ID=file_id,
+                file_grp=self.output_file_grp,
+                pageId=page_id,
+                mimetype='text/tab-separated-values',
+                local_filename=str(tsv_filepath))
+
+class OcrdNeatImportProcessor(Processor):
+
+    def __init__(self, *args, **kwargs):
+        kwargs['ocrd_tool'] = OCRD_TOOL['tools']['ocrd-neat-import']
+        kwargs['version'] = OCRD_TOOL['version']
+        super().__init__(*args, **kwargs)
+
+    def process(self):
+        """
+        Merge neat TSV results back into PAGE-XML.
+        """
+        log = getLogger('ocrd_neat.import')
+        assert_file_grp_cardinality(self.input_file_grp, 2)
+        assert_file_grp_cardinality(self.output_file_grp, 1)
+        keep_words = self.parameter['keep_words']
+        for n, (page_in_file, tsv_file) in enumerate(self.zip_input_files()):
+            page_id = page_in_file.pageId or page_in_file.ID
+            log.info('Processing: %d / %s of %d', n, page_id, len(list(self.zip_input_files())))
+            file_id = make_file_id(page_in_file, self.output_file_grp)
+            pcgts = page_from_file(self.workspace.download_file(page_in_file))
+            page = pcgts.get_Page()
+
+            tsv = pd.read_csv(tsv_file.local_filename, sep='\t', comment='#', quoting=3)
+            id_to_text = {}
+            for _, row in tsv.iterrows():
+                if str(row.TEXT).strip():
+                    id_to_text[row.line_id] = row.TEXT
+            for textline in page.get_AllTextLines():
+                if textline.id in id_to_text:
+                    textline.set_TextEquiv([TextEquivType(Unicode=id_to_text[textline.id])])
```

Comment:
How about setting `@index`? How about merely **adding** the imported text in the front? 

This could be useful to compare automatic and manual versions visually later-on (e.g. with LAREX).

Reply:
IMO this really needs to be addressed for reasons of interoperability

Reply:
Good point, we'll adress it in https://github.com/qurator-spk/page2tsv/issues/10